docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,420 @@
+___
+
+Custom Attestation Data Formats for Open Enclave
+=====
+
+This design document proposes a new attestation framework and set of APIs that
+enable developers to use custom formats for their attestation evidence and
+collateral.
+
+Motivation
+----------
+
+Currently, Open Enclave provides several APIs that developers can use for
+attestation. The two key functions are `oe_get_report`, which produces an opaque
+blob that is signed by the enclave, and `oe_verify_report`, which can be used to
+verify the generated report. The original purpose of those two APIs were to
+provide a simple, cross-platform way to produce and verify attestation data.
+
+However, for some developers, these APIs are too inflexible for their
+attestation requirements. For example, one might want to extend Open Enclave's
+current attestation structures with extra information, such as geolocation
+or a timestamp. Another user might want their enclaves to generate attestation
+data that is in a compatible format with their existing authentication
+infrastructure, such as a JSON Web Token or a X509 certificate. Overall,
+there has been interest in enchancing Open Enclave's APIs to support custom
+attestation formats to enable these scenarios.
+
```

Comment:
The need for Sergio's collateral APIs  could be added in above paragraph

Reply:
Regarding the need for collaterals API, we could add something like: "Collaterals are information--from a second source--that are used for verification of the evidence.  A new function will support getting the collaterals, while another function will support verifying the evidence along with collaterals.  The latter one, will support collaterals from any other source as long as the collateral data is in the expected format."

Reply:
Here is the updated section:
However, for some developers, these APIs are too inflexible for their
attestation requirements. For example, one might want to extend Open Enclave's
current attestation structures with extra information, such as geolocation
or a timestamp. Another user might want their enclaves to generate attestation
data that is in a compatible format with their existing authentication
infrastructure, such as a JSON Web Token or a X509 certificate. There are also
customers who want to specify their collaterals (information from a second source
used for verification, along with the evidence), instead of using the set of
collaterals provided by Open Enclave.

Overall, there has been interest in enchancing Open Enclave's APIs to support custom
attestation formats to enable these scenarios.

Reply:
Ok, I added in Sergio's updated section. I just changed "customers" to "users" so that to differentiate "product" from "project"